internal/clients/elasticsearch/security.go

```diff
@@ -292,83 +293,101 @@ func DeleteRoleMapping(ctx context.Context, apiClient *clients.ApiClient, roleMa
 	return nil
 }
 
-func PutApiKey(apiClient *clients.ApiClient, apikey *models.ApiKey) (*models.ApiKeyResponse, diag.Diagnostics) {
-	var diags diag.Diagnostics
+func CreateApiKey(apiClient *clients.ApiClient, apikey *models.ApiKey) (*models.ApiKeyCreateResponse, fwdiag.Diagnostics) {
 	apikeyBytes, err := json.Marshal(apikey)
 	if err != nil {
-		return nil, diag.FromErr(err)
+		return nil, utils.FrameworkDiagFromError(err)
 	}
 
 	esClient, err := apiClient.GetESClient()
 	if err != nil {
-		return nil, diag.FromErr(err)
+		return nil, utils.FrameworkDiagFromError(err)
 	}
 	res, err := esClient.Security.CreateAPIKey(bytes.NewReader(apikeyBytes))
 	if err != nil {
-		return nil, diag.FromErr(err)
+		return nil, utils.FrameworkDiagFromError(err)
 	}
 	defer res.Body.Close()
 	if diags := utils.CheckError(res, "Unable to create apikey"); diags.HasError() {
-		return nil, diags
+		return nil, utils.FrameworkDiagsFromSDK(diags)
 	}
 
-	var apiKey models.ApiKeyResponse
+	var apiKey models.ApiKeyCreateResponse
 
 	if err := json.NewDecoder(res.Body).Decode(&apiKey); err != nil {
-		return nil, diag.FromErr(err)
+		return nil, utils.FrameworkDiagFromError(err)
 	}
 
-	return &apiKey, diags
+	return &apiKey, nil
 }
 
-func GetApiKey(apiClient *clients.ApiClient, id string) (*models.ApiKeyResponse, diag.Diagnostics) {
-	var diags diag.Diagnostics
+func UpdateApiKey(apiClient *clients.ApiClient, apikey models.ApiKey) fwdiag.Diagnostics {
+	id := apikey.ID
+
+	apikey.Expiration = ""
+	apikey.Name = ""
+	apikey.ID = ""
```

Comment:
Does it mean that both `Name` and `Expiration` cannot be updated after creation?

Reply:
Yea, correct. They're marked as RequiresReplace in the schema as well.